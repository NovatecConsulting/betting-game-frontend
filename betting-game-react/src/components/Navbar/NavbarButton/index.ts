export { default as NavbarButton } from "./NavbarButton";
