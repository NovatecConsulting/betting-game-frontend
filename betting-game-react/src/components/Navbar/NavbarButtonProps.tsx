import { ABOUTUS, MATCHES, STANDINGS } from '../../config/routes';

export interface NavbarButtonProps {
  buttonText: string;
  buttonURL: string;
  buttonTestIdPrefix: string;
}

export const navbarButtons: NavbarButtonProps[] = [
  {
    buttonURL: MATCHES.path,
    buttonText: 'Matches',
    buttonTestIdPrefix: 'Matches',
  },
  {
    buttonURL: STANDINGS.path,
    buttonText: 'Standings',
    buttonTestIdPrefix: 'Standings',
  },
  {
    buttonURL: ABOUTUS.path,
    buttonText: 'About us',
    buttonTestIdPrefix: 'AboutUs',
  },
];
