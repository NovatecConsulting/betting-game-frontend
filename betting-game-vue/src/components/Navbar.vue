<template>
  <div class="navbar h-16 w-screen bg-gray-900 p-3 text-center overflow-hidden">
    <div class="navbar__content mx-auto flex flex-row ">
      <Logo />
      <div v-for="link in links" :key="link.name">
        <div class="navbar-link-wrapper ml-4">
          <router-link class="navbar__link relative text-xl text-blue-300 hover:text-white" :to="{ name: link.name }">
            {{ link.name }}
          </router-link>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import routes from '../router/index'
import Logo from '../components/Logo.vue'
export default {
  name: 'Homepage',
  components: {
    Logo
  },
  data() {
    return {
      links: routes.options.routes
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped></style>
