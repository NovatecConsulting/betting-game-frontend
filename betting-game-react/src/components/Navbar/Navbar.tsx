import React from "react";
import { Redirect, Route, RouteProps, Switch } from "react-router-dom";

import routes, { ABOUTUS, MATCHES, STANDINGS } from "../../config/routes";
import NavbarButton, { NavbarButtonProps } from "../NavbarButton/NavbarButton";
import NavbarLoginButton from "../NavbarLoginButton/NavbarLoginButton";
import NavbarLogo from "../NavbarLogo/NavbarLogo";

export const navbarButtons: NavbarButtonProps[] = [
    {
        buttonURL: MATCHES.path,
        buttonText: "Matches",
        buttonTestIdPrefix: "Matches",
    },
    {
        buttonURL: STANDINGS.path,
        buttonText: "Standings",
        buttonTestIdPrefix: "Standings",
    },
    {
        buttonURL: ABOUTUS.path,
        buttonText: "About us",
        buttonTestIdPrefix: "AboutUs",
    },
];

const Navbar: React.FC = () => {
    return (
        <span>
            <nav
                className="flex items-center justify-between flex-wrap border-b-2 border-blue-300 bg-blue-900 p-2"
                data-testid="Navbar"
            >
                <NavbarLogo />
                <div className="block flex-grow lg:flex lg:items-center lg:w-auto">
                    <div className="text-sm lg:flex-grow inline">
                        {navbarButtons.map(
                            (
                                navbarButton: NavbarButtonProps,
                                index: number
                            ) => (
                                <NavbarButton
                                    key={index}
                                    buttonText={navbarButton.buttonText}
                                    buttonURL={navbarButton.buttonURL}
                                    buttonTestIdPrefix={
                                        navbarButton.buttonTestIdPrefix
                                    }
                                ></NavbarButton>
                            )
                        )}
                    </div>
                </div>
                <NavbarLoginButton />
            </nav>
            <Switch>
                {routes.map((route: RouteProps, index: number) => (
                    <Route
                        data-testid="Route"
                        key={index}
                        exact
                        path={route.path}
                        component={route.component}
                    />
                ))}
                <Redirect to={MATCHES.path} />
            </Switch>
        </span>
    );
};

export default Navbar;
