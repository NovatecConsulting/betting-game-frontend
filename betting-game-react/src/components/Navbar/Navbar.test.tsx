import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './Navbar';

describe('<Navbar />', () => {
  test('it should mount', () => {
    render(<Navbar />);
    
    const navbar = screen.getByTestId('Navbar');

    expect(navbar).toBeInTheDocument();
  });
});