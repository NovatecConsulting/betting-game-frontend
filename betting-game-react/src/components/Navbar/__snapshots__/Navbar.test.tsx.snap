// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render and match snapshot with correct navbar items. 1`] = `
<span>
  <nav
    className="flex items-center justify-between flex-wrap border-b-2 border-blue-300 bg-blue-900 p-2"
    data-testid="Navbar"
  >
    <NavbarLogo />
    <div
      className="block flex-grow lg:flex lg:items-center lg:w-auto"
    >
      <div
        className="text-sm lg:flex-grow inline"
      >
        <NavbarButton
          buttonTestIdPrefix="Matches"
          buttonText="Matches"
          buttonURL="/matches"
          key="0"
        />
        <NavbarButton
          buttonTestIdPrefix="Standings"
          buttonText="Standings"
          buttonURL="/standings"
          key="1"
        />
        <NavbarButton
          buttonTestIdPrefix="AboutUs"
          buttonText="About us"
          buttonURL="/aboutus"
          key="2"
        />
      </div>
    </div>
    <NavbarLoginButton />
  </nav>
  <Switch>
    <Route
      component={[Function]}
      data-testid="Route"
      exact={true}
      key="0"
      path="/matches"
    />
    <Route
      component={[Function]}
      data-testid="Route"
      exact={true}
      key="1"
      path="/standings"
    />
    <Route
      component={[Function]}
      data-testid="Route"
      exact={true}
      key="2"
      path="/aboutus"
    />
    <Route
      component={[Function]}
      data-testid="Route"
      exact={true}
      key="3"
      path="/login"
    />
    <Redirect
      to="/matches"
    />
  </Switch>
</span>
`;
